includes/class-wc-twenty-seventeen.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+$template = get_option( 'template' );
+
+if ( 'twentyseventeen' != $template ) {
+	return; // Only do any of the things if Twenty Seventeen is the active theme
+}
+
+/**
+ * Twenty Seventeen suport.
+ *
+ * @class   WC_Twenty_Seventeen
+ * @since   2.7.0
+ * @version 2.7.0
+ * @package WooCommerce/Classes
+ */
+class WC_Twenty_Seventeen {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		remove_action( 'woocommerce_before_main_content', 'woocommerce_output_content_wrapper', 10 );
+		remove_action( 'woocommerce_after_main_content', 'woocommerce_output_content_wrapper_end', 10 );
+
+		add_action( 'woocommerce_before_main_content', array( $this, 'output_content_wrapper' ), 10 );
```

Comment:
Seems very unlikely to me.

Couldn't we just do [this](http://wordpress.stackexchange.com/a/36110) to make it easier in the unlikely event?